subprojects/docs/src/docs/userguide/jvm/java_testing.adoc

```diff
@@ -435,6 +435,66 @@ Again, we're accessing a source set to get the relevant information, i.e. where
 
+[[sec:java_testing_modular]]
+== Testing Java Modules
+
+If you are <<java_library_plugin.adoc#,developing Java Modules>>, everything described in this chapter still applies and any of the supported test frameworks can be used.
+However, there are some things to consider depending on whether you need module information to be available, and module boundaries to be enforced, during test runtime.
+In this context, often the terms _whitebox testing_ (module boundaries are deactivated or relaxed) and _blackbox testing_ (module boundaries are in place) are used.
+Whitebox testing if often used/needed for unit testing and blackbox testing often fits functional or integration test requirements.
+
+Sample: link:../samples/sample_java_modules_multi_project_with_integration_tests.html[Java Modules multi-project with integration tests]
+
+=== Whitebox unit test execution on the classpath
+
+The simplest setup to write unit tests for functions or classes in modules is to _not_ use module specifics during test runtime.
+For this, you just need to write tests the same way you would write them for normal libraries.
+If you don't have a `module-info.java` file in your test source set (`src/main/test`) this source set will be considered as traditional Java library during compilation and test runtime.
+This means, all dependencies, including Jars with module information, are put on the classpath.
+The advantage is that all internal classes of your (or other) modules are then accessible directly in tests.
+This may be a totally valid setup for unit testing, where we do not care about the larger module structure, but only about testing single functions.
+
+NOTE: If you are using Eclipse: By default, Eclipse also runs unit tests as modules using module patching (see <<#sec:java_testing_modular_patching,below>>).
+In an imported Gradle project, unit testing a module with the Eclipse test runner might fail.
+You then need to manually adjust the classpath/module path in the test run configuration or delegate test execution to Gradle.
+This only concerns the test execution.
+Unit test compilation and development works fine in Eclipse.
+
+=== Blackbox integration testing
+
+For integration tests, you have the option to define the test set itself as additional module.
+You do this similar to how you turn your main sources into a module:
+by adding a `module-info.java` file to the corresponding source set (e.g. `integrationTests/java/module-info.java`).
+
+You can find a full example that includes blackbox integration tests link:../samples/sample_java_modules_multi_project_with_integration_tests.html[here].
+
+NOTE: If you are using Eclipse: Eclipse does currently https://bugs.eclipse.org/bugs/show_bug.cgi?id=520667[not support] defining multiple modules in one project.
```

Comment:
* Use consistent NOTE formatting
* I would replace the second "Eclipse" with "it"
* currently not support -> not currently support